commit 4a57c9f95debd4ddf60147ce55602f061724a51d
Author: John Crispin <john@phrozen.org>
Date:   Wed Jan 25 09:20:55 2017 +0100

    net-next: ethernet: mediatek: change the compatible string
    
    
    [ Upstream commit 8b901f6bbcf12a20e43105d161bedde093431e61 ]
    
    When the binding was defined, I was not aware that mt2701 was an earlier
    version of the SoC. For sake of consistency, the ethernet driver should
    use mt2701 inside the compat string as this is the earliest SoC with the
    ethernet core.
    
    The ethernet driver is currently of no real use until we finish and
    upstream the DSA driver. There are no users of this binding yet. It should
    be safe to fix this now before it is too late and we need to provide
    backward compatibility for the mt7623-eth compat string.
    
    Reported-by: Sean Wang <sean.wang@mediatek.com>
    Signed-off-by: John Crispin <john@phrozen.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/ethernet/mediatek/mtk_eth_soc.c b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
index 86a89cb..4832223 100644
--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
@@ -2518,7 +2518,7 @@ static int mtk_remove(struct platform_device *pdev)
 }
 
 const struct of_device_id of_mtk_match[] = {
-	{ .compatible = "mediatek,mt7623-eth" },
+	{ .compatible = "mediatek,mt2701-eth" },
 	{},
 };
 MODULE_DEVICE_TABLE(of, of_mtk_match);
